extensions/Gemini/python/src/aiconfig_extension_gemini/Gemini.py

```diff
@@ -0,0 +1,295 @@
+
+def construct_regular_outputs(response: "AsyncGenerateContentResponse") -> list[Output]:
+    """
+    Construct regular output per response result, without streaming enabled
+    """
+    output_list = []
+    for i, candidate in enumerate(response.candidates):
+        output = ExecuteResult(
+            **{
+                "output_type": "execute_result",
+                "data": candidate.content.parts[0].text,
+                "execution_count": i,
+                "metadata": response.prompt_feedback.__dict__,
+            }
+        )
+
+        output_list.append(output)
+
+    return output_list
+
+
+async def construct_stream_outputs(
+    response: "AsyncGenerateContentResponse", options: InferenceOptions
+) -> list[Output]:
+    """
+    Construct Outputs while also streaming the response with stream callback
+
+    Gemini api seems to not support more than one candidate, so we'll just use the first one
+    """
+    output_list = []
+
+    acc = ""
+
+    async for chunk in response:
+        # deconstruct chunk
+
+        data = chunk.parts[0].text
+        acc += data
+        if options is not None and options.stream_callback is not None:
+            options.stream_callback(data, acc, 0)
+
+    output = ExecuteResult(
+        **{
+            "output_type": "execute_result",
+            "data": response.candidates[0].content.parts[0].text,
+            "execution_count": 0,  # Hard coded for now. If api supports multiple candidates this can be updated.
+            "metadata": response.prompt_feedback.__dict__,
+        }
+    )
+
+    output_list.append(output)
+
+    return output_list
+
+
+class GeminiModelParser(ParameterizedModelParser):
+    def __init__(self, id: str = "gemini-pro"):
+        super().__init__()
+        self.model = genai.GenerativeModel(id)
+
+    def id(self) -> str:
+        """
+        Returns an identifier for the model (e.g. llama-2, gpt-4, etc.).
+        """
+        return "GeminiModelParser"
+
+    async def serialize(
+        self,
+        prompt_name: str,
+        data: Dict,
+        ai_config: "AIConfigRuntime",
+        parameters: Optional[Dict] = None,
+        **kwargs,
+    ) -> List[Prompt]:
+        """
+        Defines how a prompt and model inference settings get serialized in the .aiconfig.
+
+        Args:
+            prompt (str): The prompt to be serialized.
+            inference_settings (dict): Model-specific inference settings to be serialized.
+
+        Returns:
+            str: Serialized representation of the prompt and inference settings.
+        """
+
+        event = CallbackEvent(
+            "on_serialize_start",
+            __name__,
+            {
+                "prompt_name": prompt_name,
+                "data": data,
+                "parameters": parameters,
+                "kwargs": kwargs,
+            },
+        )
+        await ai_config.callback_manager.run_callbacks(event)
+
+        raise Exception("Not Implemented")
```

Comment:
```suggestion

```

Should be removed and implemented

Reply:
@Ankush-lastmile make sure this is tracked and you don't close the GH issue until this is done.

Reply:
Left a comment on #463. This is my next high-pri focus